cmd/kpod/create.go

```diff
@@ -115,46 +135,765 @@ var createCommand = cli.Command{
+	a := getDefaultAnnotations()
+	// TODO
+	// Which annotations do we want added by default
+	if c.tty {
+		a["io.kubernetes.cri-o.TTY"] = "true"
+	}
+	return a
+}
+
+func getDefaultAnnotations() map[string]string {
+	var a map[string]string
+	a = make(map[string]string)
+	a[ann.Annotations] = ""
+	a[ann.ContainerID] = ""
+	a[ann.ContainerName] = ""
+	a[ann.ContainerType] = ""
+	a[ann.Created] = ""
+	a[ann.HostName] = ""
+	a[ann.IP] = ""
+	a[ann.Image] = ""
+	a[ann.ImageName] = ""
+	a[ann.ImageRef] = ""
+	a[ann.KubeName] = ""
+	a[ann.Labels] = ""
+	a[ann.LogPath] = ""
+	a[ann.Metadata] = ""
+	a[ann.Name] = ""
+	a[ann.PrivilegedRuntime] = ""
+	a[ann.ResolvPath] = ""
+	a[ann.HostnamePath] = ""
+	a[ann.SandboxID] = ""
+	a[ann.SandboxName] = ""
+	a[ann.ShmPath] = ""
+	a[ann.MountPoint] = ""
+	a[ann.TrustedSandbox] = ""
+	a[ann.TTY] = "false"
+	a[ann.Stdin] = ""
+	a[ann.StdinOnce] = ""
+	a[ann.Volumes] = ""
+
+	return a
+}
+
+//GetTmpfsMounts takes user provided input for bind mounts and creates Mount structs
+func (c *createConfig) GetVolumeMounts() []spec.Mount {
+	var m []spec.Mount
+	var options []string
+	for _, i := range c.volumes {
+		spliti := strings.Split(i, ":")
+		if len(spliti) > 2 {
+			options = strings.Split(spliti[2], ",")
+		}
+		// always add rbind bc mount ignores the bind filesystem when mounting
+		options = append(options, "rbind")
+		m = append(m, spec.Mount{
+			Destination: spliti[1],
+			Type:        string(TypeBind),
+			Source:      spliti[0],
+			Options:     options,
+		})
+	}
+	return m
+}
+
+//GetTmpfsMounts takes user provided input for tmpfs mounts and creates Mount structs
+func (c *createConfig) GetTmpfsMounts() []spec.Mount {
+	var m []spec.Mount
+	for _, i := range c.tmpfs {
+		// Default options if nothing passed
+		options := []string{"rw", "noexec", "nosuid", "nodev", "size=65536k"}
+		spliti := strings.Split(i, ":")
+		destPath := spliti[0]
+		if len(spliti) > 1 {
+			options = strings.Split(spliti[1], ",")
+		}
+		m = append(m, spec.Mount{
+			Destination: destPath,
+			Type:        string(TypeTmpfs),
+			Options:     options,
+		})
+	}
+	return m
+}
+
+func (c *createConfig) GetContainerCreateOptions(cli *cli.Context) ([]libpod.CtrCreateOption, error) {
+	/*
+		WithStorageConfig
+		WithImageConfig
+		WithSignaturePolicy
+		WithOCIRuntime
+		WithConmonPath
+		WithConmonEnv
+		WithCgroupManager
+		WithStaticDir
+		WithTmpDir
+		WithSELinux
+		WithPidsLimit // dont need
+		WithMaxLogSize
+		WithNoPivotRoot
+		WithRootFSFromPath
+		WithRootFSFromImage
+		WithStdin // done
+		WithSharedNamespaces
+		WithLabels //done
+		WithAnnotations // dont need
+		WithName // done
+		WithStopSignal
+		WithPodName
+	*/
+	var options []libpod.CtrCreateOption
+
+	// Uncomment after talking to mheon about unimplemented funcs
+	// options = append(options, libpod.WithLabels(c.labels))
+
+	if c.interactive {
+		options = append(options, libpod.WithStdin())
+	}
+	if c.name != "" {
+		logrus.Info("appending name %s", c.name)
```

Comment:
Do we need this logged?

Reply:
I'm OK with keeping this, but it should be dropped down to a Debugf

Reply:
Feels like a logrus.debug to me.

Reply:
fixed